deploy/application/providers.tf

```diff
@@ -58,7 +58,7 @@ data "terraform_remote_state" "infra_remote" {
   backend = "s3"
   config = {
     bucket = "tf-state-${data.aws_caller_identity.current.account_id}"
```

Comment:
Shall that backend config part be removed ? 🤔  We rely on config files anyway. It might add confusion.

Reply:
@Davidoutz Thinking out loud about it, thanks :) 

This is a `data` resource and therefore evaluated at "tf-plan" time. There, we can't access backend config files so we'd need the information somewhere else (tfvars, replaced at "compile" time, ...).

`bucket` attribute should be more flexible than `tf-state-$ACCOUNT_ID`. We have already the value in the backend config file/build time. 

We can inject values either via template replacements or tfvars. So it makes sense to remove the `bucket` valeu built like that, but using the information we already have.